src/lib/progress-spinner/progress-spinner.ts

```diff
@@ -79,31 +133,71 @@ export class MdProgressSpinner extends _MdProgressSpinnerMixinBase implements Ca
     }
   }
 
-  constructor(renderer: Renderer2, elementRef: ElementRef, platform: Platform) {
-    super(renderer, elementRef);
+  constructor(public _renderer: Renderer2, public _elementRef: ElementRef,
+              platform: Platform, @Optional() @Inject(DOCUMENT) private _document: any) {
+    super(_renderer, _elementRef);
 
     // On IE and Edge we can't animate the `stroke-dashoffset`
     // reliably so we fall back to a non-spec animation.
     const animationClass = (platform.EDGE || platform.TRIDENT) ?
       'mat-progress-spinner-indeterminate-fallback-animation' :
       'mat-progress-spinner-indeterminate-animation';
 
-    renderer.addClass(elementRef.nativeElement, animationClass);
+    _renderer.addClass(_elementRef.nativeElement, animationClass);
+  }
+
+  ngAfterViewInit() {
+    this._setDiameterClass(this.diameter);
   }
 
   ngOnChanges(changes: SimpleChanges) {
-    if (changes.strokeWidth) {
-      this._elementSize = this._baseSize + Math.max(this.strokeWidth - this._baseStrokeWidth, 0);
+    if (changes.strokeWidth || changes.diameter) {
+      this._elementSize =
+          this._diameter + Math.max(this.strokeWidth - this._baseStrokeWidth, 0);
     }
   }
 
-  _getStrokeDashOffset() {
+  get _circleRadius() {
+    return this._diameter / 2 - 5;
```

Comment:
Where does the `5` come from. I assume it's half the stroke width?

Reply:
I believe that's what it's supposed to be, yes; fixing.